### PR TITLE
Finished typescript conversions

### PR DESCRIPTION
**Pull request by bugsalexander**
Feel free to take a look! 

Note, there are still a few //todos here and there, because Jake and I decided to leave out some of the functionality stuff for another PR (still need to take out IOldRequirement, for example).

### CODE REVIEW

**Comment by JakeChvatal** [comment]
Lots of tests still appear to be failing... I'm not sure that this is ready to be merged yet.

File: package.json [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
   },
   "scripts": {
+    "lint": "tslint -c tslint.json 'src/**/*.{ts,tsx}'",
+    "pretest": "yarn lint",
     "test": "jest"
+  },
+  "dependencies": {
+    "@types/jest": "^24.0.18",
+    "tslint": "^5.19.0",
+    "typescript": "^3.6.2"
```
Comment (NilsBacke) [inline package.json]: You can move these three dependencies into `"devDependencies"`

**Comment by bugsalexander** [comment]
All tests now passing for json_parser.js!

**Comment by bugsalexander** [comment]
All tests now pass.

**Pull request merged**